src/Form/DataTransformer/ArrayToModelTransformer.php

```diff
@@ -59,11 +60,13 @@ public function reverseTransform($value): object
             return $value;
         }
 
+        $instance = Instantiator::instantiate($this->className);
+
         if (!\is_array($value)) {
-            return new $this->className();
+            return $instance;
         }
 
-        return $this->modelManager->modelReverseTransform($this->className, $value);
+        return $this->modelManager->modelReverseTransform($instance, $value);
```

Comment:
Should we rename the method ?
This could be a way to be BC and do this PR on 3.x.